app/backend/approaches/approach.py

```diff
@@ -254,6 +255,10 @@ async def compute_image_embedding(self, q: str):
         return VectorizedQuery(vector=image_query_vector, k_nearest_neighbors=50, fields="imageEmbedding")
 
     async def run(
-        self, messages: list[dict], stream: bool = False, session_state: Any = None, context: dict[str, Any] = {}
+        self,
+        messages: list[ChatCompletionMessageParam],
```

Comment:
I updated types from dict to the more precise ChatCompletionMessageParam, otherwise we get errors, since a dict can have many more properties than that particular TypedDict.